packages/compartment-mapper/src/types.js

```diff
@@ -420,12 +423,84 @@ export {};
  */
 
 /**
- * @typedef {object} Attenuator
- * @property {Function} [attenuateGlobals]
- * @property {Function} [attenuateModule]
+ * @template {[any, ...any[]]} [GlobalParams=[any, ...any[]]]
+ * @template {[any, ...any[]]} [ModuleParams=[any, ...any[]]]
+ * @typedef Attenuator
+ * @property {GlobalAttenuatorFn<GlobalParams>} [attenuateGlobals]
+ * @property {ModuleAttenuatorFn<ModuleParams>} [attenuateModule]
+ */
+
+/**
+ * @template {[any, ...any[]]} [Params=[any, ...any[]]]
+ * @callback GlobalAttenuatorFn
+ * @param {Params} params
+ * @param {Record<PropertyKey, any>} originalObject
+ * @param {Record<PropertyKey, any>} globalThis
+ * @returns {void}
+ * @todo Unsure if we can do much typing of `originalObject` and `globalThis` here.
+ */
+
+/**
+ * @template {[any, ...any[]]} [Params=[any, ...any[]]]
+ * @template [T=unknown]
+ * @template [U=T]
+ * @callback ModuleAttenuatorFn
+ * @param {Params} params
+ * @param {T} ns
+ * @returns {U}
  */
 
 /**
  * @typedef {object} DeferredAttenuatorsProvider
  * @property {(attenuatorSpecifier: string|null) => Promise<Attenuator>} import
  */
+
+/**
+ * A type representing a wildcard policy, which can be 'any'.
+ * @typedef {'any'} WildcardPolicy
+ */
+
+/**
+ * A type representing a property policy, which is a record of string keys and boolean values.
+ * @typedef {Record<string, boolean>} PropertyPolicy
+ */
+
+/**
+ * A type representing a policy item, which can be a {@link WildcardPolicy wildcard policy}, a property policy, `undefined`, or defined by an attenuator
+ * @template [T=void]
+ * @typedef {WildcardPolicy|PropertyPolicy|T} PolicyItem
+ */
+
+/**
+ * An object representing a nested attenuation definition.
+ * @typedef {Record<string, AttenuationDefinition | boolean>} NestedAttenuationDefinition
+ */
+
+/**
+ * An object representing a base package policy.
+ * @template [PackagePolicyItem=void]
+ * @template [GlobalsPolicyItem=void]
+ * @template [BuiltinsPolicyItem=void]
+ * @typedef {object} PackagePolicy
+ * @property {string} [defaultAttenuator] - The default attenuator.
```

Comment:
(nit) There's no defaultAttenuator field on the individual resource in policy in the public API. 
I'm guessing this is the consequence of defaultAttenuator being saved in the package policy of the attenuators compartment. Should we document it here? Should we sub-type internally? Should we implement a valid usecase for it?

Reply:
I saw it referenced [here](https://github.com/endojs/endo/blob/master/packages/compartment-mapper/src/policy-format.js#L132), but I can't answer your other questions.  If it's intended to be public, then great, but if not, we should subtype.

Reply:
I'd lean towards subtyping it (though I'm not sure where that type assignment needs to happen) for now unless there's an obvious use-case.